site/_js/web-components/like-icon.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview A component for displaying a "like" icon. Persist the likes
+ * to the LocalStorage and to Firebase Datastore.
+ */
+import {BaseElement} from './base-element';
+import {html} from 'lit-element';
+
+/* eslint-disable node/no-missing-import */
+// @ts-ignore
+import {initializeApp} from 'https://www.gstatic.com/firebasejs/9.6.10/firebase-app.js';
+import {
+  getDatabase,
+  connectDatabaseEmulator,
+  ref,
+  set,
+  onValue,
+  increment,
+  // @ts-ignore
+} from 'https://www.gstatic.com/firebasejs/9.6.10/firebase-database.js';
+import {
+  initializeAppCheck,
+  ReCaptchaV3Provider,
+  // @ts-ignore
+} from 'https://www.gstatic.com/firebasejs/9.6.10/firebase-app-check.js';
+
+// Local emulator configuration
+const config = {
+  apiKey: 'AIzaSyBO19CVJtI2lRZQWCjfv3FnGqStHvP4xwU',
+  projectId: 'devnook-playground-1b315',
+  appId: '1:353904485142:web:1af58ee02b74215c29a637',
+};
+const recaptchaPublicKey = '6LdDPCMfAAAAACPU7H4hn1uaV3_i9vkfIIIhX4LS';
+const storagePrefix = 'dcc-like-icon';
+
+export class LikeIcon extends BaseElement {
+  static get properties() {
+    return {
+      item: {type: String, reflect: true},
+      likes: {type: Number},
+      liked: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.item = '';
+    this.itemId = '';
+    this.onClick = this.onClick.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('click', this.onClick);
+
+    this.itemId = `${storagePrefix}-${this.item}`;
+
+    // Read like status from local storage
+    const item = localStorage.getItem(this.itemId);
+    this.liked = item === 'true';
+
+    // Read likes from Firebase DB
+    const app = initializeApp(config);
```

Comment:
Thinking about this more, should we make it so that there's a singleton of the Firebase related variables that are shared across every instance of the LikeIcon? With the current implementation, if there were multiple LikeIcons on a page (which there will be), we'll end up with a lot of repeated Firebase setup (and maybe multiple websocket connections, unless those are automatically pooled).

I'm not 100% sure what the cleanest way to implement that sort of singleton is for a web component, though.